src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/CosmosIdentityProviderExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.ControlPlane.Core.Features.Rbac;
+using Microsoft.Health.Fhir.CosmosDb.Features.Storage.ControlPlane;
+
+namespace Microsoft.Health.ControlPlane.CosmosDb.Features.Storage
+{
+    public static class CosmosIdentityProviderExtensions
+    {
+        public static IdentityProvider ToIdentityProvider(this CosmosIdentityProvider cosmosIdentityProvider)
+        {
+            return new IdentityProvider(cosmosIdentityProvider.Name, cosmosIdentityProvider.Authority, cosmosIdentityProvider.Audience)
+            {
+                Version = cosmosIdentityProvider.ETag.Trim('"'),
```

Comment:
You could make the Version property virtual in the IdentityProvider class and override it in CosmosIdentityProvider with this logic as oppose to create a new IdentityProvider from CosmosIdentityProvider.

Reply:
We still need the extension, but it's been refactored some.